### ISSUE TITLE
I'm new to this and wondering

### ISSUE THREAD

**Issue by DeceptedParasite** [LAK132/SourceExplorer#60]
how do you compile the full re-write, I've looked and can't find anything, I'm using windows if that helps

**Comment by LAK132** [comment]
you'll need git, [meson](https://mesonbuild.com/) and the latest visual studio (with build tools). 

setup:
1. clone the repo using git (just downloading it doesn't work I don't think)
2. in the command prompt `cd` into the repo
3. run `git submodule update --init --recursive`
4. run `setup.bat` 

compiling:
1. run `compile.bat` 
2. run `build\srcexp.exe`

if there are any errors feel free to let me know

**Comment by DeceptedParasite** [comment]
I've been trying for a while now and can't get meson to install. I really am not experienced with this

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129615619-aaa6666c-c9c9-4b47-a566-79a218db2c29.png)
any way you can help?

**Comment by DeceptedParasite** [comment]
it didn't come with setup.bat or compile.bat

**Comment by DeceptedParasite** [comment]
it seems to be cloning the master github branch

**Comment by LAK132** [comment]
ah, after cloning and `cd`ing in, run `git checkout 2.0-full-rewrite`

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129731567-3cc93225-1f93-4da0-944b-ad9a03bb58c0.png)
aftermath of running  setup.bat

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129734130-6f8394ac-583b-4771-a3e7-aab629a0ffb3.png
ok, so I fixed it by allowing users to change files, now I'm getting this

**Comment by LAK132** [comment]
you didn't change directory before running git clone so you've put the repo in System32, this is very bad! start from the start and make sure to change directory to the desktop or something in step 2.

**Comment by DeceptedParasite** [comment]
oh. alright

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129740787-0774d970-d206-4be7-aad2-0f03ae84af95.png)
odd. I'm getting the same message

**Comment by DeceptedParasite** [comment]
this is from start to finish

**Comment by DeceptedParasite** [comment]
and I still get the same message when running compile.bat

**Comment by DeceptedParasite** [comment]
I dont get what I'm doing wrong

**Comment by LAK132** [comment]
there's no errors in that screenshot? you need to continue with step 7

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129742359-5c93cde8-903e-427e-931f-267f74a6b1ef.png)
had trouble getting this image
thank god I got this now I can see what the problem is

**Comment by DeceptedParasite** [comment]
[meson-log.txt](https://github.com/LAK132/SourceExplorer/files/7000647/meson-log.txt)

**Comment by LAK132** [comment]
ah I've committed something I shouldn't have, give me a moment to clean that up

**Comment by DeceptedParasite** [comment]
alright

**Comment by LAK132** [comment]
from inside the repo run `git pull --ff-only` then try and compile again

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129744861-0af35e5b-2edb-45e1-85fe-993a2a5de693.png)
huh.

**Comment by LAK132** [comment]
`git pull --ff-only` and try again

**Comment by DeceptedParasite** [comment]
it's working

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129749777-2ba5804c-beb4-4f2b-8fdf-d59a80067457.png)
huh...

**Comment by LAK132** [comment]
ah sorry, committed the wrong, thing. pull and try again

**Comment by DeceptedParasite** [comment]
alright

**Comment by DeceptedParasite** [comment]
oops

**Comment by DeceptedParasite** [comment]
I'll tell you if I get the same error message again

**Comment by DeceptedParasite** [comment]
I didn't get the message this time

**Comment by DeceptedParasite** [comment]
damn, I thought we didn't need open gl anymore

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129753295-ef444d17-3157-4fdb-82f5-98bbdb8549b3.png)

**Comment by LAK132** [comment]
`build\srcexp.exe -nogl`

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129758036-b4e6d13f-194d-40ca-b3d0-a9f0457133ea.png)
dont see it

**Comment by LAK132** [comment]
run `build\srcexp.exe -nogl` from the command prompt

**Comment by DeceptedParasite** [comment]
ah, alright

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129758452-779b9cfe-1324-4950-a086-7a87dc1e40ac.png)

**Comment by DeceptedParasite** [comment]
![image](https://user-images.githubusercontent.com/64243677/129774448-88024711-7ce7-4f22-81d8-80e7a3e7e840.png)
still...

**Comment by LAK132** [comment]
did you make sure to `git pull`?

**Comment by DeceptedParasite** [comment]
I did everything in the tutorial correctly, I just dont get what I'm doing wrong

**Comment by DeceptedParasite** [comment]
and I also did 'git pull --ff-only'

**Comment by LAK132** [comment]
then you're not doing anything wrong, Source Explore is broken

**Comment by DeceptedParasite** [comment]
oh

**Comment by LAK132** [comment]
oh and make sure you run `compile.bat` again after the `git pull`

**Comment by DeceptedParasite** [comment]
yeah, I always do setup and then compile after doing git pull --ff-only

**Comment by LAK132** [comment]
your screenshot shows it being one commit old, if you're running the latest commit it should print out `Source Explorer v2.0.3-b95cf41`, maybe just try `pull`, `setup`, `compile` one more time

**Comment by DeceptedParasite** [comment]
ah, alright

**Comment by DeceptedParasite** [comment]
I did, didn't work, same message, I'mma just give up on this and wait till I get another laptop/computer, I'mma read back on this when I get one

**Comment by DeceptedParasite** [comment]
hopefully that one has openGL, the one I'm using now doesn't have openGL
